### ISSUE TITLE
Import provider.messages from WELM project

### ISSUE THREAD

**Issue by predictiple** [Velocidex/evtx#5]
https://github.com/nsacyber/Windows-Event-Log-Messages

They are missing a couple of Windows releases and their project seems to have stalled, but their repository of old stuff does provide a lot of coverage. They store a number of other useful fields too.

Unfortunately my SQL skills aren't up to the task :-(

**Comment by predictiple** [comment]
Solved via some stack overflow reading and Python! :smile_cat: 

I worked through the WELM data in more or less reverse chronological order, so that if there are any overlaps the inaccuracies will tend towards the older operating systems. Ideally one should target the specific OS version, but for bulk processing that's not always possible.

I will add data from more recent OSes, such as Win 2019, as I come across them. The message extraction & lookup is an absurdly good feature. The evtx parsing is way ahead of the pack now!

**Comment by scudette** [comment]
Heh - I looked at that project and can't believe I missed that huge zip file :-). It would be worthwhile to convert that the sqlite so we can use it as well.

Should we split into one file per release like they do or put them all in the same database?
Why not both!

**Comment by predictiple** [comment]
Suprisingly the resultant db from their 26 Vista+ OSes is only 6.6Mb and then it grew to 9Mb when I added a live extract from a Win10 machine that's not in their data. I suspect their extraction is not as thorough as dumpevtx's. 

There's plenty of duplication across all the datasets, especially since they have a lot of Windows 10 machines. So one db per release will be 26 x 2-6Mb each. I'm happy a single deduped dataset and I think there won't be any serious inaccuracies (because the core events don't change much). Busy running it now on a large set of historical evtx and it's working great!

**Comment by scudette** [comment]
I think I played with their tool while developing but unfortunately they do not provide binaries so I spent some time figuring out how to build the C# code. I noticed that they do not support MUI dlls so for cases where the messages are stored in MUI they just can not find the dll at all. 

It is a bit of a mess because sometimes there will be messages in the same dll that is pointed to in the registry but often there are no messages in that dll but the message is found by `<path>/en-us/<dll>.mui` files. Sometimes there are actually messages in the base dll and also messages (for the same event id) in the MUI and sometimes those messages different even say different things for the same event ID!

Sigh - this is the trouble with the whole scheme - it leads to messy logging that is hard to work with. Our life as sysadmins and security people is hard enough we don't need to be fighting the system too!

**Comment by scudette** [comment]
I would love to host it but I dont want to put it in this Git repo - we use it as a dependency for Velociraptor and i dont want to have to download mbs of sqlite when we dont need to. 

Maybe make another repo?

**Comment by predictiple** [comment]
I've emailed you a link to the file. I think most people are going to be running the live extraction and have little interest in offline enrichment, but for me it's a huge win.

When I added data from the live Windows 10 machine to the WELM db I can anecdotally confirm that what you're saying about MUI sources. The data that dumpevtx was able to add all seemed to come from less conventional sources.
